test/bitcoin_cln_test.go

```diff
@@ -1141,3 +1141,77 @@ func Test_ClnCln_ExcessiveAmount(t *testing.T) {
 	})
 
 }
+
+// Test_ClnCln_StuckChannels tests that the swap fails if the channel is stuck.
+// For more information about stuck channel, please check the link.
+// https://github.com/lightning/bolts/issues/728
+func Test_ClnCln_StuckChannels(t *testing.T) {
+	IsIntegrationTest(t)
+	t.Parallel()
+
+	require := require.New(t)
+	// repro by using the push_msat in the open_channel.
+	// Assumption that feperkw is 253perkw in reg test.
+	bitcoind, lightningds, scid := clnclnSetupWithConfig(t, 3794, 3573, []string{
```

Comment:
Numbers checked. They match 1% reserve 37940 msat and a CommitTx fee of 183172 msat.
This gives the funder 221112 msat for a stuck channel. The configuration leaves 221 sat on the funders side which results in a stuck channel.